jax/core.py

```diff
@@ -92,54 +91,46 @@ def jaxprs_in_params(params) -> Iterator[Jaxpr]:
     for v in vals:
       if isinstance(v, Jaxpr):
         yield v
-      elif isinstance(v, TypedJaxpr):
+      elif isinstance(v, ClosedJaxpr):
         yield v.jaxpr
 
 
 def subjaxprs(jaxpr: Jaxpr) -> Iterator[Jaxpr]:
   """Generator for all subjaxprs found in the params of jaxpr.eqns.
+
   Does not descend recursively into the found subjaxprs.
   """
   for eqn in jaxpr.eqns:
     yield from jaxprs_in_params(eqn.params)
 
 
-class TypedJaxpr:
+class ClosedJaxpr:
   jaxpr: Jaxpr
-  literals: List['Any']
-  in_avals: List['AbstractValue']
-  out_avals: List['AbstractValue']
+  consts: List['Any']
 
-  def __init__(self, jaxpr: Jaxpr, literals: Sequence,
-               in_avals: Sequence['AbstractValue'],
-               out_avals: Sequence['AbstractValue']):
-    assert len(literals) == len(jaxpr.constvars)
-    assert len(in_avals) == len(jaxpr.invars)
+  def __init__(self, jaxpr: Jaxpr, consts: Sequence):
+    assert len(consts) == len(jaxpr.constvars)
+    self.jaxpr = jaxpr
+    self.consts = list(consts)
 
-    if not skip_checks:
```

Comment:
Being able to remove this invariant check is a powerful reason to make the change.